pkg/storage/ingest/writer_test.go

```diff
@@ -643,10 +645,15 @@ func TestWriter_WriteSync(t *testing.T) {
 		assert.GreaterOrEqual(t, numPartitions, 10)
 		assert.Equal(t, 1, cfg.WriteClients)
 
-		t.Cleanup(func() {
-			t.Log("releasing produce requests")
-			close(unblockProduceRequests)
-		})
+		doUnblockProduceRequests := func() {
+			unblockProduceRequestsOnce.Do(func() {
+				t.Log("releasing produce requests")
+				close(unblockProduceRequests)
+			})
+		}
+
+		// Ensure produce requests are released in case of prematurely test termination.
```

Comment:
[Nit]
```suggestion
		// Ensure produce requests are released in case of premature test termination.
```

Reply:
Done in https://github.com/grafana/mimir/pull/8738/commits/c74391199b1270bc2f9bdb5ae55c1ca79c18d30e